### ISSUE TITLE
How to integrate LevelDB?

### ISSUE THREAD

**Issue by doraemondrian** [mafintosh/hypercore#139]
Hi, newbie here. Thanks for creating this project. Coming from working with IPFS the simplicity of this library is a breath of fresh air. I appreciate the effort that went into keeping it simple.

I just hoped you guys would have more centralized branding so more people can find out about this project (There's a huge disparity between what DAT/Hypercore/Hyperdrive can do vs. what it's positioned as) because I have known DAT for a while but never bothered to check out hypercore until now.

Anyway, here's the real question: I "rediscovered" this project through this wonderful conference speech at https://www.youtube.com/watch?v=7tf14VpeHlE

I had read some of the documentation before watching the video so I was surprised to find that you can just require `level` and pass it into `hypercore` to create an object, because that's what I was trying to do.

But when I actually tried the code presented in the video it didn't work. I presume the syntax has changed or you guys dropped some APIs?

More specifically, when I tried to:

```
var core = hypercore(level('test.db'))
```

it fails with:

```
  if (typeof createStorage !== 'function') throw new Error('Storage should be a function or string')
                                           ^

Error: Storage should be a function or string
```

So I looked deeper into the code, and looking at the API as of today, looks like it resembles a lot the syntax of `level` itself, for example you instantiate with:

```
var feed = hypercore('./db', {valueEncoding: 'utf-8'})
```

So my first guess was maybe you guys integrated leveldb natively since then, so looked into the code but couldn't find it.

So far I think I figured out that I'm supposed to pass a function in the form of https://github.com/random-access-storage/random-access-storage but I'm not sure how LevelDB fits into this and would appreciate some pointer.

TLDR: How can i simply use leveldb with hypercore? Thank you so much!

**Comment by noffle** [comment]
Hi Dora. hypercore uses the random-access-storage family[1] of modules,
which is wholly unrelated to LevelDB.

[1] https://github.com/random-access-storage

**Comment by doraemondrian** [comment]
@noffle thank you for the answer. Does this mean the API has changed? At least in the video I saw that the presenter directly passed a leveldb instance into hypercore and the demo just worked.

I am aware of https://github.com/mafintosh/hyperdb and may end up using it but was just confused about what's going on.

**Comment by noffle** [comment]
I looked at the git log, and it looks like it was indeed level based! It seems to have changed back in 3e3f082fec77639e52c89737409c072923389f09 to the new file format.

**Comment by mafintosh** [comment]
In the very first versions we were leveldb backed but it was much, much slower than the current flat file approach
